hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test/AccountBalanceSigningUtilsTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.cli.sign.test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.hedera.services.cli.sign.AccountBalanceSigningUtils;
+import java.io.File;
+import java.nio.file.Path;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.Objects;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+
+class AccountBalanceSigningUtilsTest {
+    @Mock
+    PublicKey publicKey;
+
+    @Mock
+    PrivateKey privateKey;
+
+    @TempDir
+    private File tmpDir;
+
+    @Test
+    @DisplayName("Failed to generate signature file with invalid KeyPair")
+    void failureGenerateSignatureFileInvalidKeyPair() {
+        // given:
+        final var signatureFileDestination = Path.of("testPath");
+        final var keyPair = new KeyPair(publicKey, privateKey);
+        final var fileToSign = Path.of(Objects.requireNonNull(AccountBalanceSigningUtilsTest.class
+                        .getClassLoader()
+                        .getResource("com.hedera.services.cli.sign.test/2023-03-23T14_25_29.576022Z_Balances.pb"))
+                .getPath());
+
+        // then:
+        assertThrows(
+                RuntimeException.class,
+                () -> AccountBalanceSigningUtils.signAccountBalanceFile(signatureFileDestination, fileToSign, keyPair));
+    }
+
+    @Test
+    @DisplayName("Succeed to generate signature file")
+    void generateSignatureFile() {
+        // given:
+        final var signatureFileDestination = Path.of(tmpDir.getPath() + "/2023-03-23T14_25_29.576022Z_Balances.pb_sig");
+        final var fileToSign = Path.of(Objects.requireNonNull(AccountBalanceSigningUtilsTest.class
+                        .getClassLoader()
+                        .getResource("com.hedera.services.cli.sign.test/2023-03-23T14_25_29.576022Z_Balances.pb"))
+                .getPath());
+
+        // then:
+        assertTrue(AccountBalanceSigningUtils.signAccountBalanceFile(
```

Comment:
Is there anything here or in another test to validate that the resulting signature file is actually _correct_?

This test suite is generally pretty good - so if that's missing, it shouldn't be!  (Maybe I just didn't spot it.)

Reply:
Yes, I added unit tests to test matching original signed files.